core/storage/collection/gae_models.py

```diff
@@ -434,6 +433,69 @@ def export_data(cls, user_id):
         }
 
 
+class CollectionRightsAllUsersModel(base_models.BaseModel):
+    """Temporary storage model for all user ids ever mentioned in the collection
+    rights.
+
+    TODO (#8529): This model should be deleted after the user ID migration is
+    completed.
+
+    The id of each instance is the id of the corresponding collection.
+    """
+    # The user_ids of users who are (or were in history) members of owner_ids,
+    # editor_ids, voice_artist_ids or viewer_ids in corresponding rights model.
+    all_user_ids = ndb.StringProperty(indexed=True, repeated=True)
```

Comment:
We should have a validation function in prod_validation to ensure that these ids correspond to valid UserSettingsModel.

Reply:
Done.